### PR TITLE
Lobby join & part & some session persistence fixes

### PR DESCRIPTION
**Pull request by infernalfire72**

### CODE REVIEW

File: app/packet_handlers.py [hunk 1]
```diff
@@ -1309,6 +1384,12 @@ async def user_leaves_channel_handler(session: "Session", packet_data: bytes):
     channel = await channels.fetch_one_by_name(channel_name)
     if channel is None:
         return
+    
+    # NOTE: we ignore #lobby to enqueue the match updates
+    # and we actually remove them from the channel on lobby part
+    presence: Presence | None = session.get("presence")
```
Comment (tsunyoku) [inline app/packet_handlers.py]: is there a reason get is used? presence should never be None

Comment (cmyui) [reply app/packet_handlers.py]: yeah this is true for now, will adjust

File: .github/workflows/staging-deploy.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - INTRN-168-lobbied
```
Comment (cmyui) [inline .github/workflows/staging-deploy.yml]:
```suggestion
```

**Review by cmyui — APPROVED** [review]
+1 👍🏻

**Review by cmyui — APPROVED** [review]
+1 👍🏻

**Pull request merged**